src/utils/useLocationUpdateEffect.ts

```diff
@@ -33,6 +33,24 @@ const useLocationUpdateEffect = (
 
     return () => clearInterval(intervalId);
   }, [user, roomName]);
+
+  useEffect(() => {
+    if (!user || !roomName) return;
+
+    const firestore = firebase.firestore();
+    const doc = `users/${user.uid}/visits/${roomName}`;
+    const increment = firebase.firestore.FieldValue.increment(1);
+
+    const intervalId = setInterval(() => {
+      return firestore
+        .doc(doc)
+        .update({ timeSpent: increment })
+        .catch(() => {
+          firestore.doc(doc).set({ timeSpent: 1 });
+        });
+    }, 5 * 1000);
```

Comment:
secondsSpent = timeSpent * 5? This is pretty confusing, please use constants, comments etc to make this clear.